core/src/word_lock.rs

```diff
@@ -154,7 +154,7 @@ impl WordLock {
                 if let Err(x) = self.state.compare_exchange_weak(
                     state,
                     state.with_queue_head(thread_data),
-                    Ordering::Release,
+                    Ordering::AcqRel,
```

Comment:
Nice catch! Thinking about it for a bit, an acquire is indeed needed here so that the call to `park` happens after the thread is inserted into the queue.